packages/core/src/node/keytar-server.ts

```diff
@@ -0,0 +1,98 @@
+// code copied and modified from https://github.com/microsoft/vscode/blob/1.55.2/src/vs/platform/native/electron-main/nativeHostMainService.ts#L679-L771
+
+import { KeytarService } from '../common/keytar-protocol';
+import { injectable } from 'inversify';
+import { isWindows } from '../common';
+import * as keytar from 'keytar';
+
+@injectable()
+export class KeytarServiceImpl implements KeytarService {
+    private static readonly MAX_PASSWORD_LENGTH = 2500;
+    private static readonly PASSWORD_CHUNK_SIZE = KeytarServiceImpl.MAX_PASSWORD_LENGTH - 100;
+
+    async setPassword(service: string, account: string, password: string): Promise<void> {
+        if (isWindows && password.length > KeytarServiceImpl.MAX_PASSWORD_LENGTH) {
+            let index = 0;
+            let chunk = 0;
+            let hasNextChunk = true;
+            while (hasNextChunk) {
+                const passwordChunk = password.substring(index, index + KeytarServiceImpl.PASSWORD_CHUNK_SIZE);
+                index += KeytarServiceImpl.PASSWORD_CHUNK_SIZE;
+                hasNextChunk = password.length - index > 0;
+
+                const content: ChunkedPassword = {
+                    content: passwordChunk,
+                    hasNextChunk: hasNextChunk
+                };
+
+                await keytar.setPassword(service, chunk ? `${account}-${chunk}` : account, JSON.stringify(content));
+                chunk++;
+            }
+
+        } else {
+            await keytar.setPassword(service, account, password);
+        }
+    }
+
+    deletePassword(service: string, account: string): Promise<boolean> {
+        return keytar.deletePassword(service, account);
+    }
+
+    async getPassword(service: string, account: string): Promise<string | undefined> {
+        const password = await keytar.getPassword(service, account);
+        if (password) {
+            try {
+                let { content, hasNextChunk }: ChunkedPassword = JSON.parse(password);
+                if (!content || !hasNextChunk) {
+                    return password;
+                }
+
+                let index = 1;
+                while (hasNextChunk) {
+                    const nextChunk = await keytar.getPassword(service, `${account}-${index++}`);
+                    const result: ChunkedPassword = JSON.parse(nextChunk!);
+                    content += result.content;
+                    hasNextChunk = result.hasNextChunk;
+                }
+
+                return content;
+            } catch {
+                return password;
+            }
+        }
+    }
+    async findPassword(service: string): Promise<string | undefined> {
+        const password = await keytar.findPassword(service);
+        if (password) {
+            return password;
+        }
+    }
+    async findCredentials(service: string): Promise<Array<{ account: string, password: string }>> {
+        return [];
```

Comment:
maybe remove `findCredentials` or leave a comment explaining why we need to keep it

Reply:
Fixed by adding a real implementation